examples/favorites_advanced/getx_favorites_advanced/lib/feature_puppy/search/views/search_page.dart

```diff
@@ -1,30 +1,45 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:get/get.dart';
-
-import 'package:getx_favorites_advanced/feature_puppy/favorites/controllers/favorite_puppies_controller.dart';
+import 'package:getx_favorites_advanced/feature_puppy/controllers/puppy_extra_details_controller.dart';
+import 'package:getx_favorites_advanced/feature_puppy/controllers/puppy_list_controller.dart';
+import 'package:favorites_advanced_base/ui_components.dart';
 
 class SearchPage extends StatelessWidget {
   @override
-  Widget build(BuildContext context) => Container(
-      constraints: BoxConstraints(minWidth: MediaQuery.of(context).size.width),
-      child: Column(
-        mainAxisAlignment: MainAxisAlignment.center,
-        crossAxisAlignment: CrossAxisAlignment.center,
-        children: [
-          const Text(
-            'SEARCH PAGE',
-            style: TextStyle(fontSize: 20),
-          ),
-          ElevatedButton(
-            onPressed: Get.find<FavoritePuppiesController>().incrementCount,
-            child: const Text('Increment Favorites'),
-          ),
-          ElevatedButton(
-            onPressed: Get.find<FavoritePuppiesController>().decrementCount,
-            child: const Text('Decrement Favorites'),
-          )
-        ],
-      ),
+  Widget build(BuildContext context) {
+    final controller = Get.find<PuppyListController>();
+    return Obx(() => controller.isLoading.value
+        ? LoadingWidget()
+        : controller.hasError.value
+            ? const ErrorRetryWidget(
+                onReloadTap: _onReload,
+              )
+            : Obx(
+                () => ListView.builder(
+                  key: const Key('LoadingWidget'),
```

Comment:
@RomanovaPrime I believe we have Keys as a class. So please use it here.

Reply:
I've created a new Keys file in my base folder, where to put keys that are not included in base project. When we merge my branch to develop, I can move them in base project.

Reply:
Didn't get that, can you elaborate?